src/main/java/seedu/address/model/article/UniqueArticleList.java

```diff
@@ -165,4 +170,45 @@ private boolean articlesAreUnique(List<Article> articles) {
         }
         return true;
     }
+
+    /**
+     * Looks up the article with the given articleID in the article book.
+     * Returns a list of persons who are related to the article.
+     */
+    public List<Person> lookupArticle(String articleID, AddressBook addressBook) throws CommandException {
+        requireAllNonNull(articleID, addressBook);
+        Article article = this.lookupArticle(articleID);
+        List<Person> persons = new ArrayList<>();
+
+        for (Author authors : article.getAuthors()) {
+            for (Person person : addressBook.getPersonList()) {
+                if (person.getName().fullName.equals(authors.authorName)) {
+                    persons.add(person);
+                }
+            }
+        }
+        for (Source sources : article.getSources()) {
+            for (Person person : addressBook.getPersonList()) {
+                if (person.getName().fullName.equals(sources.sourceName)) {
+                    persons.add(person);
+                }
+            }
+        }
+        article.setPersonList(persons);
```

Comment:
Since space was already allocated to each `Article` object noting down all related `Persons` the first time the `lookupArticle` method is called, if we make it mandatory to do all these looping and checking for `Persons` each time we are looking up an `article` that has already been looked up, this may be flagged as a feature flaw since we are unnecessarily performing these operations instead of referring to whats already stored in the `article`.